gpflow/posterior.py

```diff
@@ -503,7 +510,8 @@ def _conditional_with_precompute(
         cov = Kff - Kfu_Qinv_Kuf
 
         if not full_cov and not full_output_cov:
-            cov = tf.linalg.adjoint(cov)
+            if cov.shape.ndims > 1:
```

Comment:
in what case would this ever be 1?
second question- how will this interact with tf.function() (and potentially dynamic tensor shapes)?

Reply:
Using a `FullyCorrelatedPosterior` with a `SharedIndependent` kernel.

Reply:
hm that seems like a bug... I would've always expected it to have shape [M, 1] at least...

Reply:
This seems to have been addressed by ensuring that `Qinv` has an `L` axis.